### PR TITLE
show wanings on page when codesnippet not found

### PR DESCRIPTION
**Pull request by yishengjin1413**
Refer to this feature: https://ceapex.visualstudio.com/Engineering/_workitems/edit/20848/.
Spec is here: https://review.docs.microsoft.com/en-us/new-hope/specs/ops/missing-sample-build?branch=master#design

We want to show a warning on page when the codesnippt is not found.

### CODE REVIEW

File: src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs [hunk 1]
```diff
@@ -17,6 +17,10 @@ public class HtmlCodeSnippetRenderer : HtmlObjectRenderer<CodeSnippet>
     {
         private readonly MarkdownContext _context;
         private const string tagPrefix = "snippet";
+        private const string warningMessageId = "codeIncludeNotFound";
+        private const string defaultWarningMessage = "It looks like the sample you are looking for has moved!  Rest assured we are working on resolving this.";
```
Comment (yufeih) [inline src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs]: Are you going to modify template to include these messages?

Comment (yishengjin1413) [reply src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs]: Yes. My change have checked in and localized: https://github.com/Microsoft/templates.docs.msft.zh-cn/blob/master/ContentTemplate/token.json#L75

Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs]: The wording is good for docs user, but not well applies to community user. What about "It looks like the sample you are looking for does not exist." as default message?

**Comment by superyyrrzz** [comment]
Will you also change dfm engine?

**Comment by yishengjin1413** [comment]
@superyyrrzz I think we don't need to update DFM since it will retired soon and I remember we have stop supporting DFM long time ago, right?  Just keep the DFM have no breaking change for it is ok.

File: src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs (hunk 1 above)
Comment (yishengjin1413) [reply src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs]: Oh, so for en-us build, template will also pass the token to markdown engine? If that, I think it's make sense to use the message you mentioned as the default string.

Comment (superyyrrzz) [reply src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs]: Yes, it exists: https://github.com/Microsoft/templates.docs.msft/blob/master/ContentTemplate/token.json#L75

**Pull request merged**